File: app/bot/BotImpl.scala
package bot
import game.Controller
import telegram.TelegramApi

import scala.concurrent.Future

class BotImpl(chatId: Long, controller: Controller, api: TelegramApi) extends Bot {

  override def sendMessage(text: String): Future[Unit] = {
    api.sendText(chatId, text)
  }

  override def sendMusic(filePath: String): Future[Unit] = {
    api.sendMusic(chatId, filePath)
  }

  override def finishGame(): Future[Unit] = Future.successful {
    controller.finishGame(chatId)
  }
}

File: app/game/BotImpl.java
package game;

/**
 * Created by krotkov on 7/30/2016.
 */
public class BotImpl implements Bot {
    long chatId;
    Controller controller;

    public BotImpl(long chatId, Controller controller) {
        this.chatId = chatId;
        this.controller = controller;
    }

    @Override
    public void sendMessage(String text) {
        TelegramAPI.sendText(chatId, text);
    }

    @Override
    public void sendMusic(String filePath) {
        TelegramAPI.sendMusic(chatId, filePath);
    }

    @Override
    public void finishGame() {
        controller.finishGame(chatId);
    }
}
